libsolidity/ast/Types.cpp

```diff
@@ -2224,6 +2232,24 @@ u256 StructType::storageSize() const
 	return max<u256>(1, members(nullptr).storageSize());
 }
 
+bool StructType::canLiveOutsideStorage() const
+{
+	bool result = true;
+
+	for (auto const& member: m_struct.members())
+	{
+		// Avoids infinite loops of `canLiveOutsideStorage()` calls. For recursive structs that
```

Comment:
What about two structs that refer to each other?

Reply:
Good point. I need to think about recursion again.

This currently this goes into an infinite loop:
```solidity
contract C {
	struct S {
		T[] t;
	}

	struct T {
		S[] s;
	}

	function f() {
		S memory s;
	}
}
```